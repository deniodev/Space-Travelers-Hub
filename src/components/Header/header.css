.nav-bar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: whitesmoke;
  padding: 0.5rem 2rem;
}

.logo-content {
  display: flex;
  align-items: center;
  gap: 2rem;
}

.logo-content p {
  font-size: 1.5rem;
}

.logo {
  width: 100px;
}

.logo img {
  width: 100%;
}

.nav-link {
  display: flex;
  align-items: center;
  gap: 2rem;
  font-size: 1.5rem;
}

.nav-link a {
  text-decoration: none;
  color: black;
}

a.active {
  text-decoration: underline;
}
