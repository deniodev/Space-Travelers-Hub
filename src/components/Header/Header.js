import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../../images/planet.png';
import './header.css';

const Header = () => (
  <nav className="nav-bar">
    <div className="logo-content">
      <div className="logo">
        <img src={logo} alt="logo" />
      </div>
      <p>Space Traveler&apos;s Hub</p>
    </div>
    <ul className="nav-link">
      <NavLink to="/">Rockets</NavLink>
      <NavLink to="mission">Missions</NavLink>
      <NavLink to="profile">My profile</NavLink>
    </ul>
  </nav>
);

export default Header;
