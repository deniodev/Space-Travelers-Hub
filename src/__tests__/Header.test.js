import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header/Header';

describe('tests for Navigation menu', () => {
  test('snapshot for Navbar component', () => {
    const component = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
