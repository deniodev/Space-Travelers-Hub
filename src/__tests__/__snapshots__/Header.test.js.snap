// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Navigation menu snapshot for Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav-bar"
      >
        <div
          class="logo-content"
        >
          <div
            class="logo"
          >
            <img
              alt="logo"
              src="planet.png"
            />
          </div>
          <p>
            Space Traveler's Hub
          </p>
        </div>
        <ul
          class="nav-link"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
          <a
            class=""
            href="/profile"
          >
            My profile
          </a>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav-bar"
    >
      <div
        class="logo-content"
      >
        <div
          class="logo"
        >
          <img
            alt="logo"
            src="planet.png"
          />
        </div>
        <p>
          Space Traveler's Hub
        </p>
      </div>
      <ul
        class="nav-link"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Rockets
        </a>
        <a
          class=""
          href="/missions"
        >
          Missions
        </a>
        <a
          class=""
          href="/profile"
        >
          My profile
        </a>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
